src/main/java/org/mule/extension/http/api/policy/HttpListenerPolicyParametersTransformer.java

```diff
@@ -66,33 +66,37 @@ public Message fromFailureResponseParametersToMessage(Map<String, Object> parame
 
   @Override
   public Map<String, Object> fromMessageToSuccessResponseParameters(Message message) {
-    return messageToResponseParameters(new HttpListenerSuccessResponseBuilder(), "response", message);
+    return messageToResponseParameters(new HttpListenerSuccessResponseBuilder(), "response", message, 200);
   }
 
   @Override
   public Map<String, Object> fromMessageToErrorResponseParameters(Message message) {
-    return messageToResponseParameters(new HttpListenerErrorResponseBuilder(), "errorResponse", message);
+    return messageToResponseParameters(new HttpListenerErrorResponseBuilder(), "errorResponse", message, 500);
   }
 
   private Map<String, Object> messageToResponseParameters(HttpListenerResponseBuilder httpListenerResponseBuilder,
-                                                          String responseBuilderParameterName, Message message) {
+                                                          String responseBuilderParameterName, Message message,
+                                                          int defaultStatusCode) {
     ImmutableMap.Builder<String, Object> mapBuilder =
         ImmutableMap.<String, Object>builder().put(responseBuilderParameterName, httpListenerResponseBuilder);
     if (message.getAttributes().getValue() instanceof HttpResponseAttributes) {
       HttpResponseAttributes httpResponseAttributes = (HttpResponseAttributes) message.getAttributes().getValue();
       httpListenerResponseBuilder.setBody(message.getPayload());
       httpListenerResponseBuilder.setHeaders(httpResponseAttributes.getHeaders());
-      httpListenerResponseBuilder.setStatusCode(httpResponseAttributes.getStatusCode());
+      httpListenerResponseBuilder.setStatusCode(httpResponseAttributes.getStatusCode() == 0 ? defaultStatusCode
```

Comment:
So this depends on statusCode remaining undefined? Wouldn't it be best to distinguished success from error attributes and give those the defaults? Or even better, settling on 200 and redefining it to 500 in case of errors.